rct229/rules/section5.py

```diff
@@ -97,3 +111,215 @@ def rule_check(self, context, calc_vals, data=None):
+                        or scc_b == "SEMI-EXTERIOR"
+                    ):
+                        target = table_G34_lookup(climate_zone, scc_b, BELOW_GRADE_WALL)
+                        target_c_factor = (
+                            target["c_factor"] if "c_factor" in target else None
+                        )
+
+                        if target_c_factor is None:
+                            raise Exception(
+                                f"Failed finding the c-factor for below grade wall in climate zone {climate_zone}"
+                            )
+
+                    elif scc_b == "EXTERIOR MIXED":
+                        target = table_G34_lookup(
+                            climate_zone, "EXTERIOR RESIDENTIAL", BELOW_GRADE_WALL
+                        )
+                        target_c_factor_res = (
+                            target["c_factor"] if "c_factor" in target else None
+                        )
+
+                        target = table_G34_lookup(
+                            climate_zone, "EXTERIOR NON-RESIDENTIAL", BELOW_GRADE_WALL
+                        )
+                        target_c_factor_nonres = (
+                            target["c_factor"] if "c_factor" in target else None
+                        )
+
+                        if (
+                            target_c_factor_res is None
+                            or target_c_factor_nonres is None
+                        ):
+                            raise Exception(
+                                f"Failed finding the c-factor for below grade wall in climate zone {climate_zone}"
+                            )
+
+                        if (
+                            target_c_factor_res.magnitude
+                            != target_c_factor_nonres.magnitude
+                        ):
+                            mix_surface_c_factor_ids.append(surface_b["id"])
+                        else:
+                            target_c_factor = target_c_factor_res
+                    # convert values to IP unit to compare with standard.
+                    model_c_factor_magnitude = round(
+                        surface_construction_b["c_factor"]
+                        .to("Btu_h / square_foot / delta_degF")
+                        .magnitude,
+                        2,
+                    )
+                    target_c_factor_magnitude = target_c_factor.magnitude
+                    if model_c_factor_magnitude != target_c_factor_magnitude:
+                        failing_surface_c_factor_ids.append(surface_b["id"])
+
+            calc_val["failed_c_factor_surface_id"] = failing_surface_c_factor_ids
+            calc_val["mix_surface_c_factor_ids"] = mix_surface_c_factor_ids
+            return calc_val
+
+        def rule_check(self, context, calc_vals, data=None):
+            return len(calc_vals["failed_c_factor_surface_id"]) == 0
+
+        def manual_check_required(self, context, calc_vals=None, data=None):
```

Comment:
I'd like to propose that we put `manual_check_required` above `rule_check` since it comes before it in our execution logic.